packages/components/src/font-size-picker/utils.ts

```diff
@@ -44,24 +44,20 @@ const FONT_SIZES_ALIASES = [
 ];
 
 /**
- * Helper util to split a font size to its numeric value
- * and its `unit`, if exists.
+ * Given a font size, returns the numeric part and unit part. For example, given
+ * '30px', 30 and 'px' would be returned.
  *
  * @param  size Font size.
- * @return An array with the numeric value and the unit if exists.
+ * @return An array with the number and the unit if it exists.
  */
-export function splitValueAndUnitFromSize(
-	size: NonNullable< FontSizePickerProps[ 'value' ] >
-) {
-	const [ numericValue, unit ] = `${ size }`.match( /[\d\.]+|\D+/g ) ?? [];
-
-	if (
-		! isNaN( parseFloat( numericValue ) ) &&
-		isFinite( Number( numericValue ) )
-	) {
-		return [ numericValue, unit ];
+export function parseNumberAndUnitFromSize(
```

Comment:
Not really necessary for this PR but I thought I'd update this function to:

- accept `undefined`, since that's a valid input to `FontSizePicker`;
- be more strict about what is returned, i.e.
  - specify a return type;
  - return a `number` instead of a `string` containing a number;
- use a regular regex with capturing groups instead of relying on `/g` behaviour;
- not use "value" in the name since that can be confused with the `value` prop in `FontSizePicker`.

Reply:
`UnitControl` has got a bunch of utils function that potentially achieve a very similar (if not the same) result: https://github.com/WordPress/gutenberg/blob/1326a46cf10ad54e897d1f3c9aaa0e385aba896a/packages/components/src/unit-control/utils.ts

Maybe we could use those function directly? Also happy if we want to leave this task separate and work on it in the future.

Reply:
Oh fantastic! That lets us delete this altogether 😀